exec/plugin-commands-script-runners/src/dlx.ts

```diff
@@ -81,7 +81,7 @@ export async function handler (
     await add.handler({
       // Ideally the config reader should ignore these settings when the dlx command is executed.
       // This is a temporary solution until "@pnpm/config" is refactored.
-      ...omit(['workspaceDir', 'rootProjectManifest'], opts),
+      ...(omit(['workspaceDir', 'rootProjectManifest', 'symlink'] as Array<keyof DlxCommandOptions>, opts) as DlxCommandOptions),
```

Comment:
This is terribly hacky.  The problem is `handler` is pretending that `opts` only contains the keys of `DlxCommandOptions`, but, in actuality, other keys are also present, such as `symlink`.

I could add `'symlink'` to [`Pick<Config, ...>`](https://github.com/pnpm/pnpm/blob/7cd0d20bb27e49354187cacce3fdd4ae090127f2/exec/plugin-commands-script-runners/src/dlx.ts#L67), but that doesn't seem right.

I think a proper typechecked solution would be for `handler` to recognize that `opts` can be anything, and to use a `pick` function instead `omit`.  But that would be a larger change.

Or, if looser typechecking is fine, we could use an `omit` function that accepts any keys and returns `Partial<DlxCommandOptions>`.

Thoughts?

Reply:
> I could add 'symlink' to [Pick<Config, ...>](https://github.com/pnpm/pnpm/blob/7cd0d20bb27e49354187cacce3fdd4ae090127f2/exec/plugin-commands-script-runners/src/dlx.ts#L67), but that doesn't seem right.

you can do this